### PR TITLE
Remove native code from stride.core project

### PR DESCRIPTION
**Pull request by Jklawreszuk**
# PR Details

This PR tries to remove native files from the entire Stride.Core project

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sources/engine/Stride.Native/NativeInvoke.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ internal class Module
         [ModuleInitializer]
         public static void Initialize()
         {
-            Core.Native.NativeInvoke.Setup();
+            
         }
```
Comment (Eideren) [inline sources/engine/Stride.Native/NativeInvoke.cs:47]: Probably fine to remove it entirely, no ?

**Review by Eideren — COMMENTED** [review]
Looks good. Did you have the time to check that linux allocator in the same folder ?

**Comment by Jklawreszuk** [comment]
> Looks good. Did you have the time to check that linux allocator in the same folder ?

@Eideren As far as I can see, the LinuxAllocator.cpp file is currently being used by Stride.Audio and Stride.Navigation, so currently we can't get rid of this file.

File: sources/engine/Stride.Native/NativeInvoke.cs (hunk 1 above)
Comment (Jklawreszuk) [reply sources/engine/Stride.Native/NativeInvoke.cs]: Good idea. I removed it and it seems to work anyway

**Comment by Eideren** [comment]
@Jklawreszuk it sounds like we could probably replace it with `Marshal.AllocHGlobal` or something similar, though I haven't looked at how those are used
Thanks !

**Pull request merged**